contracts/crowdfund/ETHCrowdfundBase.sol

```diff
@@ -219,72 +224,88 @@ abstract contract ETHCrowdfundBase is Implementation {
-                payable(msg.sender).transferEth(refundAmount);
-            }
+        if (newTotalContributions > maxTotalContributions_) {
+            revert ExceedsRemainingContributionsError(
+                contribution,
+                maxTotalContributions_ - totalContributions
+            );
         } else {
             totalContributions = newTotalContributions;
-        }
 
-        // Check that the contribution amount is at or above the minimum. This
-        // is done after `amount` is potentially reduced if refunding excess
-        // contribution. There is a case where this prevents a crowdfunds from
-        // reaching `maxTotalContributions` if the `minContribution` is greater
-        // than the difference between `maxTotalContributions` and the current
-        // `totalContributions`. In this scenario users will have to wait until
-        // the crowdfund expires or a host finalizes after
-        // `minTotalContribution` has been reached by calling `finalize()`.
-        uint96 minContribution_ = minContribution;
-        if (amount < minContribution_) {
-            revert BelowMinimumContributionsError(amount, minContribution_);
+            if (
+                maxTotalContributions_ == newTotalContributions ||
+                minContribution_ > maxTotalContributions_ - newTotalContributions
+            ) {
+                _finalize(newTotalContributions);
+            }
         }
 
-        // Subtract fee from contribution amount if applicable.
-        address payable fundingSplitRecipient_ = fundingSplitRecipient;
-        uint16 fundingSplitBps_ = fundingSplitBps;
-        if (fundingSplitRecipient_ != address(0) && fundingSplitBps_ > 0) {
-            // Removes funding split from contribution amount in a way that
-            // avoids rounding errors for very small contributions <1e4 wei.
-            amount = (amount * (1e4 - fundingSplitBps_)) / 1e4;
+        // Check that the contribution amount is at or above the minimum.
+        if (contribution < minContribution_) {
+            revert BelowMinimumContributionsError(contribution, minContribution_);
         }
 
         // Calculate voting power.
-        votingPower = (amount * exchangeRateBps) / 1e4;
+        votingPower = convertContributionToVotingPower(contribution);
 
         if (votingPower == 0) revert ZeroVotingPowerError();
     }
 
+    /// @notice Calculate the voting power amount that would be received from
+    ///         the given contribution.
+    /// @param contribution The contribution amount.
+    /// @return votingPower The voting power amount that would be received from
+    ///                     the contribution.
+    function convertContributionToVotingPower(
+        uint96 contribution
+    ) public view returns (uint96 votingPower) {
+        contribution = _removeFundingSplitFromContribution(contribution);
+        votingPower = _convertContributionToVotingPower(contribution);
+    }
```

Comment:
Very confusing having two functions with essentially the same name doing different math